common/scala/src/main/scala/org/apache/openwhisk/core/containerpool/ContainerFactory.scala

```diff
@@ -52,7 +52,23 @@ case class ContainerPoolConfig(userMemory: ByteSize, concurrentPeekFactor: Doubl
  */
 trait ContainerFactory {
 
-  /** create a new Container */
+  /**
+   * Create a new Container
+   *
+   * The created container has to satisfy following requirements:
+   * - The container's file system is based on the provided action image and has a read/write layer on top.
+   * - If the specified image is not available on the system, it is pulled from an image
+   *   repository - for example, Docker Hub.
+   * - The container has a network interface which is able to connect to the internet
+   *   (in particular, to the specified DNS servers). In addition, the invoker must be able
+   *   to connect to the container's interface.
+   * - The IPv4 address of said interface is stored in the created Container instance.
```

Comment:
Are we really constrained to IPv4 here? I also don't think that it's necessarily a requirement to store said address in the Container.

Reply:
Well, the `Container` trait has built-in connection management based on the stored `ContainerAddress`. The trait also has implementations for `initialize()` and `run()` based on `callContainer()`. You could overwrite all that code - but I'm talking about the default.

I guess we never really tried whether IPv6 would work...

Proposal:

> The IPv4 address of said interface is stored in the created Container instance if you want to use the standard init / run behaviour defined in the Container trait.

Reply:
I'm not even positive that it's always an ipv4 address. Let's just simplify to address and go with that?

Reply:
Replaced `IPv4 address` with `IP address`.